tasks/WebRTCReceiveTask.cpp

```diff
@@ -167,15 +167,19 @@ void WebRTCReceiveTask::handleVideoStream(GstElement* bin, GstPad* pad)
     GstElement* q = gst_element_factory_make("queue", NULL);
     GstElement* conv = gst_element_factory_make("videoconvert", NULL);
     GstElement* sink = gst_element_factory_make("appsink", "video_out");
+    g_object_set(sink, "max-buffers", 1, NULL);
+    g_object_set(sink, "drop", true, NULL);
 
-    gst_bin_add_many(GST_BIN(bin), q, conv, sink, NULL);
-    gst_element_sync_state_with_parent(q);
+    // gst_bin_add_many(GST_BIN(bin), q, conv, sink, NULL);
+    gst_bin_add_many(GST_BIN(bin), conv, sink, NULL);
+    // gst_element_sync_state_with_parent(q);
     gst_element_sync_state_with_parent(conv);
     gst_element_sync_state_with_parent(sink);
-    gst_element_link_many(q, conv, sink, NULL);
+    // gst_element_link_many(q, conv, sink, NULL);
+    gst_element_link_many(conv, sink, NULL);
     configureOutput(bin, "video_out", _frame_mode.get(), false, _video_out);
 
-    GstPad* qpad = gst_element_get_static_pad(q, "sink");
+    GstPad* qpad = gst_element_get_static_pad(conv, "sink");
```

Comment:
Remove the queue *completely*.  It's still created (memory leak) and you have the commented-out code that was adding it.